packages/Interface/src/DTK_C_API.cpp

```diff
@@ -311,26 +314,33 @@ const char *DTK_git_commit_hash()
 
 DTK_Handle DTK_create( DTK_ExecutionSpace space )
 {
-    return new DataTransferKit::DTK_Struct( space );
+    auto handle = new DataTransferKit::DTK_Registry( space );
+    DataTransferKit::valid_handles.insert( handle );
+    return handle;
 }
 
-void DTK_destroy( DTK_Handle handle )
+bool DTK_is_valid( DTK_Handle handle )
 {
-    auto dtk = reinterpret_cast<DataTransferKit::DTK_Struct *>( handle );
+    return DataTransferKit::valid_handles.count( handle );
+}
 
-    if ( dtk )
+void DTK_destroy( DTK_Handle handle )
+{
+    if ( DataTransferKit::valid_handles.count( handle ) )
+    {
+        auto dtk = reinterpret_cast<DataTransferKit::DTK_Registry *>( handle );
```

Comment:
Because `DTK_Handle` is typedefed to `void*` the answer is "YES we probably should be using `static_cast` but had we done something like
```
typedef struct { } *DTK_Handle;
```
then we would need the `reinterpret_cast`.